src/app/app.module.ts

```diff
@@ -35,4 +40,86 @@ import { PostsComponent } from './posts/posts.component';
   bootstrap: [AppComponent]
 })
 
-export class AppModule { }
+export class AppModule { 
+  defineErrorsFromResponse(res: any, requiredProperty: any, module: Component): any
+  {
+    if (!res || res.status == undefined) {
+      if (requiredProperty) {
+        module.error = 'none';
+        return false;
+      }
+
+      module.error = 'error';
+      module.error_message = res;
+      return true;
+    }
+
+    if (res.status >= 200 && res.status < 300) {
+      module.error = 'none';
+      return false;
+    }
+
+    switch (res.status) {
+      case 401:
+        module.error = 'unauthorized';
+        break;
+      case 0:
```

Comment:
cases that are the same thing y can just pile the cases like 

case x:
case y :
same code